### ISSUE TITLE
Use NNPDF as a dependency

### ISSUE THREAD

**Issue by ElieHammou** [HEP-PBSP/SIMUnet#107]
We want to move away from the NNPDF fork that SIMUnet relies upon currently to use the current NNPDF version as a dependency. We would only modify/add the functions which are specific to SIMUnet. We need to identify SIMUnet-specific functions and either:
- Add them if they are SIMUnet specific function (SIMUnet label), or
- Replace/merge them carefully if they are shared with NNPDF (NNPDF label).

---

## From `validphys`:

- `validphys/simunet_analysis.py`  
  _Contains SMEFT post-fit analysis and plotting tools._  
  This file should probably be kept as is.

- `validphys/bsmnames.py`  
  _Builds names of SMEFT K-factors from runcard info._  
  This file should probably be kept as is.

- `level0_commondata_wc` from `validphys/pseudodata.py`  
  _Generates level 0 pseudodata and adds SMEFT K-factors to SM predictions._  
  We need to merge our modifications with the current NNPDF version if it has changed.

---

### Functions in `config.py`:

- `parse_dataset_input` *NNPDF*
- `produce_dataset` *NNPDF*
- `produce_fitinputcontext` *NNPDF*
- `produce_bsm_fac_initialisations` *SIMUnet*
- `produce_bsm_names_to_latex` *SIMUnet*
- `produce_bsm_names_to_plot_scales` *SIMUnet*
- `produce_contamination_data` *SIMUnet*
- `produce_n_simu_parameters` *SIMUnet*
- `produce_simu_parameters_linear_combinations` *SIMUnet*
- `produce_simu_parameters_names` *SIMUnet*
- `produce_simu_parameters_scales` *SIMUnet*

---

### Functions in `core.py`:

- From class `DataSetInput`:
  - `__init__` *NNPDF*

- From class `DataSetSpec`:
  - `__init__` *NNPDF*
  - `load` *NNPDF*

- From class `FitSpec`:
  - `as_input` *NNPDF*

---

### Functions in `loader.py`:

- From class `Loader`:
  - `check_dataset` *NNPDF*
  - `check_fktable` *NNPDF*
  - `get_simu_parameters_name_dict` *SIMUnet*

---

### Functions in `fitdata.py`:

- `_check_has_bsm_facs` *SIMUnet*
- `read_bsm_facs` *SIMUnet*
- Global constant: `BSM_FAC_FILE = 'bsm_fac.csv'` *SIMUnet*
- Global variable: `pdfs_fits_read_read_bsm_facs = collect(...)` *SIMUnet*

---

### Functions in `n3fit_data_utils.py`:

- `common_data_reader_dataset` *SIMUnet*
- `parse_simu_parameters_names_CF` *SIMUnet*

---

### Functions in `n3fit_data.py`:

- `fitting_data_dict` *SIMUnet*

---

### Functions in `results.py`:

- From class `ThPredictionsResult`:
  - `__init__` *NNPDF*
  - `from_convolution` *NNPDF*

- Top-level functions:
  - `__init__` *NNPDF*
  - `dataset_inputs_results` *NNPDF*
  - `from_convolution` *NNPDF*
  - `one_or_more_results` *NNPDF*
  - `pdf_results` *NNPDF*
  - `results` *NNPDF*
  - `dataset_bsm_factor` *SIMUnet*
  - `dataset_inputs_bsm_factor` *SIMUnet*

---

### Functions in `vp_comparefits.py`:

- `add_positional_arguments` *NNPDF*
- `complete_mapping` *NNPDF*
- `get_commandline_arguments` *NNPDF*

---

## From `n3fit`:

- `validphys/CombineCfac.py`  
  _Creates the combination layer of SIMUnet_

### Functions in `model_gen.py `:

- `_generate_experimental_layer` *NNPDF*
 _This function builds the extra SMEFT layer in the neural network_

---

### Functions in `model_trainer.py`:

- `__init__` *NNPDF*
- `_generate_observables` *NNPDF*
- `hyperparametrizable` *NNPDF*

---

### Functions in `perform_fit.py`:

- `performfit` *NNPDF*

**Comment by scarlehoff** [comment]
Starting goal, `core.py`, `loader.py`, `config.py` -> get something from which you can do `check_dataset` `API.dataset` and get the simunet versions.

1. `class SimuNetLoader(Loader)` -> `def _check_theory_old_or_new(self ... )` (`return super().<name of method>`)

that function gives you from a) FkTables b) Commondata -> `fkspec`

that function will call `check_fk_from_theory_metadata` and `check_cfac` to produce fktables and cfactors, so it should be enough with overriding `def check_fk_from_theory_metadata`.


Note: in simunet you were using length(cfactor array) == length(data), 4 fktables -> 4 cfactors

"tests":
a) -> check_dataset() -> should work ootb
b) -> check_fktable() -> more complicated, you should see a `FkTableSpec` that knows about your simunet cfactors already https://github.com/NNPDF/nnpdf/blob/fef95e6d490656c5c789efb8dcc0f3b1fc0088ff/validphys2/src/validphys/core.py#L570


----
Monkey patching https://github.com/NNPDF/nnpdf/blob/fef95e6d490656c5c789efb8dcc0f3b1fc0088ff/validphys2/examples/API_extension_Pineappl.ipynb
